botocore/parsers.py

```diff
@@ -386,10 +386,13 @@ def _do_error_parse(self, response, shape):
         root = self._parse_xml_string_to_dom(xml_contents)
         parsed = self._build_name_to_xml_node(root)
         self._replace_nodes(parsed)
-        # Once we've converted xml->dict, we need to make one more
-        # adjustment to be consistent with ResponseMetadata
-        # for non-error responses:
-        # {"RequestId": "id"} -> {"ResponseMetadata": {"RequestId": "id"}}
+        # Once we've converted xml->dict, we need to make one or two
+        # more adjustments to extract nested errors and to be consistent
+        # with ResponseMetadata for non-error responses:
+        # 1. {"Errors": {"Error": {...}}} -> {"Error": {...}}
+        # 2. {"RequestId": "id"} -> {"ResponseMetadata": {"RequestId": "id"}}
+        if 'Errors' in parsed and isinstance(parsed['Errors'], dict):
```

Comment:
But we know what the service response is.  Unless we have a concrete example of a service doing this, I'm against hypothetical these cases.  There's _many_ places in the parsing code where we make strong assumptions about how an AWS service responds.